regolith/chained_db.py

```diff
@@ -0,0 +1,17 @@
+"""Base class for chaining DBs"""
+
+from collections import ChainMap
+
+
+class ChainDB(ChainMap):
+    """ A ChainMap who's ``_getitem__`` returns an instance of itself"""
```

Comment:
Since getitem does not return `self` (or a copy of self) it is not fair to say that it returns itstance of itself.  Rather, it returns an instance of ChainDB.  Also should it return an instance of ChainMap?

Also, should there be a setitem method so that incoming mappings are converted to chainmaps?